apps/backend/src/core/shipment/infrastructure/web/shipment.controller.ts

```diff
@@ -1,13 +1,32 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
-import { Controller, Get, HttpStatus, Param } from "@nestjs/common";
+import { BadRequestException, Controller, Get, Param } from "@nestjs/common";
 import { CommandBus } from "@nestjs/cqrs";
 
+import { CountryEnum } from "src/core/port/domain/enums/country.enum";
+import { validateCountries } from "src/utils/country.utils";
+
+import { FindBestRouteCommand } from "../../application/commands/find-best-route/find-best-route.command";
+import { FindBestRouteResponseDto } from "../../application/commands/find-best-route/find-best-route.response.dto";
+
 @Controller("shipment")
 export class ShipmentController {
   constructor(private readonly commandBus: CommandBus) {}
 
   @Get("/find-best-route/:countryDeparture/:countryArrival")
-  async findBestRoute(@Param("countryDeparture") countryDeparture: string, @Param("countryArrival") countryArrival: string) {
-    return HttpStatus.OK;
+  async findBestRoute(
+    @Param("countryDeparture") countryDeparture: string,
+    @Param("countryArrival") countryArrival: string,
+  ): Promise<FindBestRouteResponseDto> {
+    const error = validateCountries(countryDeparture, countryArrival);
```

Comment:
**suggestion:** Here this is typically business logic leakage 😬 . It is not in your controller (infra/web) that you want to code what kind of country is supported by your app.

This is the responsability of your application layer, to properly check business rules (ie: "do we support that country ?") and to properly throw "business" errors.

Then those business errors should be catched by your calling stack, in that case the controller, that should react properly to the business error, by throwing itself an appropriate HTTP error.